test_app/integration_tests/android_mocks/src/main/java/com/voxeet/sdk/services/ConferenceService.java

```diff
@@ -0,0 +1,243 @@
+package com.voxeet.sdk.services;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.voxeet.VoxeetSDK;
+import com.voxeet.android.media.errors.SpatialArgumentException;
+import com.voxeet.android.media.errors.SpatialAudioException;
+import com.voxeet.android.media.spatialisation.SpatialDirection;
+import com.voxeet.android.media.spatialisation.SpatialPosition;
+import com.voxeet.android.media.spatialisation.SpatialScale;
+import com.voxeet.promise.Promise;
+import com.voxeet.sdk.json.ParticipantInfo;
+import com.voxeet.sdk.json.ParticipantInvited;
+import com.voxeet.sdk.models.Conference;
+import com.voxeet.sdk.models.Participant;
+import com.voxeet.sdk.models.ParticipantPermissions;
+import com.voxeet.sdk.services.builders.ConferenceCreateOptions;
+import com.voxeet.sdk.services.builders.ConferenceJoinOptions;
+import com.voxeet.sdk.services.builders.VideoForwardingOptions;
+import com.voxeet.sdk.services.conference.AudioProcessing;
+import com.voxeet.sdk.utils.Opt;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import kotlin.Unit;
+
+public class ConferenceService {
+
+    @Nullable
+    private Conference currentConference = null;
+
+    private String mConferenceId = null;
+
+    @NotNull
+    public Promise<Conference> create(@NotNull ConferenceCreateOptions conferenceCreateOption) {
+        return Promise.resolve(new Conference());
+    }
+
+    @Nullable
+    public Conference getConference() {
+        return currentConference;
+    }
+
+    @Nullable
+    public String getConferenceId() {
+        return mConferenceId;
+    }
+
+    @NonNull
+    public Conference getConference(@NonNull String conferenceId) {
+        return new Conference();
+    }
+
+    @NotNull
+    public Promise<Conference> fetchConference(@NotNull String conferenceId) {
+        return Promise.resolve(currentConference);
+    }
+
+    @NotNull
+    public Promise<Conference> join(@NonNull ConferenceJoinOptions options)  {
+        String id = Opt.of(options).then(ConferenceJoinOptions::getConferenceId).or("");
+        return new Promise<>(solver -> {
+            solver.resolve(currentConference);
+        });
+    }
+
+    public Promise<Boolean> leave() {
+        return Promise.resolve(true);
```

Comment:
We need a way of "remote controlling" these kind of results. Please look at the iOS mocks for inspiration :-)